cuda/matrix/ell_kernels.cu

```diff
@@ -167,11 +168,56 @@ void advanced_spmv(std::shared_ptr<const CudaExecutor> exec,
 GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_ELL_ADVANCED_SPMV_KERNEL);
 
+namespace kernel
+{
+
+template <typename ValueType>
+__global__ void initialize_zero_dense(size_type num_rows, size_type num_cols,
+		size_type stride, ValueType* result)
+{
+	auto tidx_x = threadIdx.x + blockDim.x * blockIdx.x;
+	auto tidx_y = threadIdx.y + blockDim.y * blockIdx.y;
+	if (tidx_x < num_rows && tidx_y < num_cols) {
+		result[tidx_x * stride + tidx_y] = zero<ValueType>();
+	}
+}
+
+template <typename ValueType, typename IndexType>
+__global__ __launch_bounds__(default_block_size) void fill_in_dense(size_type num_rows, size_type nnz, size_type source_stride, const IndexType* col_idxs,
+		const ValueType* values, size_type result_stride, ValueType* result)
+{
+	auto tidx = threadIdx.x + blockIdx.x * blockDim.x;
+	if (tidx < num_rows) {
+		for (auto col = 0; col < nnz; col++) {
+			result[tidx * result_stride + col_idxs[tidx + col * source_stride]] +=
+					values[tidx + col * source_stride];
+		}
+	}
+}
+
+} // namespace kernel
 
 template <typename ValueType, typename IndexType>
 void convert_to_dense(
     std::shared_ptr<const CudaExecutor> exec, matrix::Dense<ValueType> *result,
-    const matrix::Ell<ValueType, IndexType> *source) NOT_IMPLEMENTED;
+    const matrix::Ell<ValueType, IndexType> *source)
+{
+	auto num_rows = result->get_size()[0];
+	auto num_cols = result->get_size()[1];
+	auto result_stride = result->get_stride();
+	auto col_idxs = source->get_const_col_idxs();
+	auto vals = source->get_const_values();
+	auto source_stride = source->get_stride();
+
+	dim3 init_block_size = (32, 32, 1);
+	dim3 init_grid_dim = (ceildiv(num_rows, 32), ceildiv(num_cols, 32), 1);
```

Comment:
All variables should be `const`.
Also: remove the `=` while calling a constructor.